velox/dwio/parquet/reader/ParquetTypeWithId.cpp

```diff
@@ -51,19 +51,19 @@ bool ParquetTypeWithId::hasNonRepeatedLeaf() const {
 }
 
 LevelMode ParquetTypeWithId::makeLevelInfo(LevelInfo& info) const {
-  int16_t repeatedAncestor = 0;
-  for (auto parent = parquetParent(); parent;
-       parent = parent->parquetParent()) {
-    if (parent->type()->kind() == TypeKind::ARRAY ||
-        parent->type()->kind() == TypeKind::MAP) {
-      repeatedAncestor = parent->maxDefine_;
-      break;
-    }
-  }
+
   bool isList = type()->kind() == TypeKind::ARRAY;
   bool isStruct = type()->kind() == TypeKind::ROW;
   bool isMap = type()->kind() == TypeKind::MAP;
   bool hasList = false;
+  int16_t repeatedAncestor = maxDefine_;
+  auto node = this;
+  do {
+    if (node->isOptional_) {
```

Comment:
Roughly based on https://github.com/apache/arrow/blob/main/cpp/src/parquet/level_conversion.h#L125. 
the do() is to make sure that for nested types we are going inside  and checking optional since the child layers optional (which we ignored) is needed by arrow